app/test/unit/git/stash-test.ts

```diff
@@ -25,42 +27,119 @@ describe('git/stash', () => {
     })
 
     it('returns all stash entries created by Desktop', async () => {
-      await generateTestStashEntries(repository)
+      await generateTestStashEntry(repository, 'master', false)
+      await generateTestStashEntry(repository, 'master', false)
+      await generateTestStashEntry(repository, 'master', true)
 
       const stashEntries = await getDesktopStashEntries(repository)
 
       expect(stashEntries).toHaveLength(1)
       expect(stashEntries[0].branchName).toBe('master')
     })
   })
+
+  describe('createStashEntry', () => {
+    let repository: Repository
+    let readme: string
+
+    beforeEach(async () => {
+      repository = await setupEmptyRepository()
+      readme = path.join(repository.path, 'README.md')
+      await FSE.writeFile(readme, '')
+      await GitProcess.exec(['add', 'README.md'], repository.path)
+      await GitProcess.exec(['commit', '-m', 'initial commit'], repository.path)
+    })
+
+    it('creates a stash entry', async () => {
+      await await FSE.appendFile(readme, 'just testing stuff')
+      const tipSha = await getTipSha(repository)
+
+      await createStashEntry(repository, 'master', tipSha)
+
+      const result = await GitProcess.exec(['stash', 'list'], repository.path)
+      const entries = result.stdout.trim().split('\n')
+      expect(entries).toHaveLength(1)
+    })
+  })
+
+  describe('getLastStashEntry', () => {
+    let repository: Repository
+    let readme: string
+
+    beforeEach(async () => {
+      repository = await setupEmptyRepository()
+      readme = path.join(repository.path, 'README.md')
+      await FSE.writeFile(readme, '')
+      await GitProcess.exec(['add', 'README.md'], repository.path)
+      await GitProcess.exec(['commit', '-m', 'initial commit'], repository.path)
+    })
+
+    it('returns null when no stash entries exist for branch', async () => {
+      await generateTestStashEntry(repository, 'some-other-branch', true)
+
+      const entry = await getLastStashEntry(repository, 'master')
+
+      expect(entry).toBeNull()
+    })
+
+    it('returns last entry made for branch', async () => {
+      const branchName = 'master'
+      await generateTestStashEntry(repository, branchName, true)
+      const lastEntry = await generateTestStashEntry(
+        repository,
+        branchName,
+        true
+      )
+
+      const actual = await getLastStashEntry(repository, branchName)
+
+      expect(actual!.stashSha).toBe(lastEntry)
+    })
+  })
 })
 
-async function stash(repository: Repository, message?: string) {
+async function getTipSha(repository: Repository) {
   const result = await GitProcess.exec(['rev-parse', 'HEAD'], repository.path)
-  const tipSha = result.stdout.trim()
+  return result.stdout.trim()
+}
+
+async function stash(
+  repository: Repository,
+  branchName: string,
+  message: string | null
+): Promise<string> {
+  const tipSha = await getTipSha(repository)
+  const result = await GitProcess.exec(['stash', 'create'], repository.path)
```

Comment:
I need the object Id of the stash entry because I'm asserting based on that in test for `getLastStashEntry`

Reply:
💭 if you find yourself writing comments to explain the intent of a function, maybe these could live as comments in the source (either as API documentation or inside the function, whichever help to explain the intent better)